spring-cloud-open-service-broker-core/src/main/java/org/springframework/cloud/servicebroker/exception/ServiceBrokerMaintenanceInfoConflictException.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.servicebroker.exception;
+
+/**
+ * Thrown to indicate that the {@literal maintenance_info.version} provided in the request does not match the {@literal
+ * maintenance_info.version} described for the service plan in the service broker's catalog
+ *
+ * <p>
+ * Throwing this exception will result in an HTTP status code {@literal 422 UNPROCESSABLE ENTITY} being returned to the
+ * platform.
+ *
+ * @author Roy Clarkson
+ */
+public class ServiceBrokerMaintenanceInfoConflictException extends ServiceBrokerException {
+
+	private static final long serialVersionUID = 3665795988607316116L;
+
+	/**
+	 * Error code representing that asynchronous processing is required by the service broker
+	 */
+	public final static String MAINTENANCE_INFO_CONFLICT_ERROR = "MaintenanceInfoConflict";
+
+	/**
+	 * Construct an exception with the provided message.
+	 *
+	 * @param message the exception message
+	 */
+	public ServiceBrokerMaintenanceInfoConflictException(String message) {
+		super(MAINTENANCE_INFO_CONFLICT_ERROR, "Service broker maintenance info conflict: " + message);
```

Comment:
I know it says MAY, but shall we return the message suggested by the spec? 
```
"The maintenance information for the requested Service Plan has changed.
```
https://github.com/openservicebrokerapi/servicebroker/blob/v2.15/spec.md#response-3

Reply:
yeah, good point. Since we're returning a message anyway, might as well return the one they suggest.